grails-app/services/org/pih/warehouse/data/ProductSupplierDataService.groovy

```diff
@@ -133,15 +133,15 @@ class ProductSupplierDataService {
     }
 
     def getOrCreateNew(Map params) {
-        if (ProductSupplier.get(params.productSupplier.id) != null) {
-            return ProductSupplier.get(params.productSupplier.id)
+        if (ProductSupplier.get(params.productSupplier) != null) {
+            return ProductSupplier.get(params.productSupplier)
         }
 
         Product product = Product.get(params.product.id)
         ProductSupplier productSupplier = new ProductSupplier()
-        productSupplier.code = params.productSupplier.id?:params.supplierCode
+        productSupplier.code = params.productSupplier?:params.supplierCode
         productSupplier.supplierCode = params.supplierCode
-        productSupplier.name = params.supplierCode
+        productSupplier.name = params.sourceName ?: params.supplierCode
```

Comment:
I think we should ask kelsey about this, but I think this should probably be one of two options:

Option A
```
productSupplier.name = params.sourceName ?: product?.name
```
Option B
```
productSupplier.name = params.sourceName ?: null
```
I believe Kelsey mentioned that she would prefer Option B, but let's ask her.

Reply:
Kelsey says use product.name so let's go with Option A.